awq/models/base.py

```diff
@@ -153,21 +154,38 @@ def from_quantized(self, model_path, model_type, model_filename='',
         device_map = infer_auto_device_map(
             model,
             no_split_module_classes=[self.layer_type], 
+            max_memory=max_memory,
             dtype=torch_dtype
         )
 
         # Load checkpoint
         load_checkpoint_in_model(
             model,
             checkpoint=model_weights_path,
-            device_map=device_map
+            device_map=device_map,
+            offload_folder=offload_folder,
+            dtype=torch_dtype
         )
         
         # Dispath to devices
-        model = simple_dispatch_model(model, device_map)
+        if max_memory is None:
+            # VRAM only
+            model = simple_dispatch_model(model, device_map)
+
+            if fuse_layers:
+                self.fuse_layers(model, quant_config)
+        else:
+            if fuse_layers:
+                self.fuse_layers(model, quant_config)
+
+            # Offloading dispatch
+            from accelerate import dispatch_model
+            model = dispatch_model(
+                model,
+                device_map=device_map,
+                offload_dir=offload_folder
+            )
```

Comment:
Why not just use `dispatch_model()` instead if it works?

Reply:
You're right. It works just fine. I've removed call to the `simple_dispatch()`. And fixed a minor bug in fused/attn.py